conversation/io.yaktor.conversation.cs/src/io/yaktor/validation/ConversationValidator.xtend

```diff
@@ -39,6 +40,32 @@ class ConversationValidator extends AbstractConversationValidator {
     }
     // TODO: disallow t.causedBy.name == t.triggers.name
   }
+  
+  @Check
+  def checkValidEventTypeContinuity(Transition t) {
+    var cause = t?.causedBy
+    var trigger =t?.triggers ?: t?.exTriggers
+    if (!t.requiresExecution && cause != null && cause?.refType !=  trigger.refType) {
+      warning("Input doesn't match output.", t,
+        ConversationPackage.eINSTANCE.transition_CausedBy, IssueCodes.TRANSITION_TYPE_IMPEDENCE)
+    }
+  }
+  
+  @Check
+  def checkValidEventMapping(Transition t) {
+    if (t.mapping == null && t.exCausedBy?.refType != t?.toState?.parent?.parent?.projection) {
+      warning("Cause implies mapping.", t,
+        ConversationPackage.eINSTANCE.transition_CausedBy, IssueCodes.TRANSITION_TYPE_IMPEDENCE)
+    }
+  }
+  
+  @Check
+  def checkValidEventType(Transition t) {
```

Comment:
Issue #18's item 3 has two parts.  This looks like the code for the first part, but the second reads "change code so that scope provider for a Transition's mapping only provides list of t.publiclyReceives.refType's fields".  I don't see changes for that in this PR.